### DISCUSSION TITLE
build.gradle setup pulling latest wpilib from repo?

### DISCUSSION THREAD

**Discussion by jasonli0616** [mjansen4857/pathplanner#312] [Q&A] [answered]
It appears that `CommandBase` was deprecated [2 days ago in a commit](https://github.com/wpilibsuite/allwpilib/commit/aaea85ff165602a612e57a0e8170eaaa43ceec8f) to `main` on wpilib. However, I have version 2023.4.3 installed locally (latest stable release, from March this year).

For some reason, VSCode is now showing `CommandBase` as deprecated on my computer, even though I haven't updated wpilib (no released were even made since this change, documentation hasn't been updated) and it has broken the code, specifically in `pathplannerlib/src/main/java/com/pathplanner/lib/auto/BaseAutoBuilder.java`.

No releases have been made since the commit deprecating this; is the build.gradle somehow getting the latest wpilib code from the wpilib repo?

![Screenshot 2023-07-16 at 12 01 06 PM](https://github.com/mjansen4857/pathplanner/assets/75756042/ae18039e-6b6b-4ad1-aa23-6fc784f9a09a)
![Screenshot 2023-07-16 at 12 01 54 PM](https://github.com/mjansen4857/pathplanner/assets/75756042/82ca9883-968d-405f-88a9-6810d0838fb4)

**Comment by mjansen4857** [comment]
You'll likely want to ask over at the WPILib repo about this. Is your version of GradleRIO in build.gradle hardcoded to a specific version (should be line 3)?

**Reply by mjansen4857** [reply]
Also, does code still build or is there an error while building? Its possible that it could just be pulling docs for the main branch for some reason.

**Reply by jasonli0616** [reply]
There's an error while building (main branch on this repo), CommandBase was deprecated and the methods defined in Commands now return Command rather than CommandBase.

This was changed in a commit 2 days ago in wpilib with no release made so I'm not sure how that's possible.

**Comment by jasonli0616** [comment]
@mjansen4857 The dependencies in the `build.gradle` are pulling the latest HEAD rather than the latest stable release. Changes were made on the `allwpilib` repo which were not updated in a release, which caused the issue.

```gradle
implementation 'edu.wpi.first.wpilibNewCommands:wpilibNewCommands-java:2023.+'
```

https://github.com/mjansen4857/pathplanner/blob/fc33c444ca6c6a070a499559d49521f168e8a636/pathplannerlib/build.gradle#L28-L46

Resolved, thanks

**Reply by mjansen4857** [reply]
That should only affect making new pplib builds, not what you have on your end. That will also only pull actual releases.

No WPILib code gets included with PPLib builds, so it uses whatever WPILib build you're running.

**Reply by jasonli0616** [reply]
Could it be the release mode, with `-Preleasemode`?

https://github.com/mjansen4857/pathplanner/blob/fc33c444ca6c6a070a499559d49521f168e8a636/pathplannerlib/build.gradle#L18-L22

**Reply by jasonli0616** [reply]
I have 2023.4.3 installed on my computer, however running `./gradlew publish` and `./gradlew publish -Preleasemode` in the `pathplannerlib` directory on `main` causes issues with the deprecation of `CommandBase`, which hasn't been included in any releases.

**Reply by mjansen4857** [reply]
PPLib builds are built with release mode:

https://github.com/mjansen4857/pathplanner/blob/fc33c444ca6c6a070a499559d49521f168e8a636/.github/workflows/pplib-ci.yml#L52C45-L52C45

Even if they weren't, no new PPLib builds have been released since the end of the season. So, there is something going on with your end of WPILib stuff that is causing it to pull development builds. I just tried building a robot project myself and all is working fine for me. If you ran `./gradlew build -PreleaseMode` for your robot project, does that solve your issue? If so, I'm not sure what would be causing a robot project to use dev builds.

**Reply by mjansen4857** [reply]
Building pathplannerlib locally with `./gradlew build -PreleaseMode` also works fine for me.

**Reply by jasonli0616** [reply]
I'm attempting to build `pathplannerlib`, not a robot project, sorry for the confusion. It's telling me that `CommandBase` has been marked for deprecation, and is throwing errors related to that. I will reinstall WPILib to see if something is wrong with my current installation.

**Reply by mjansen4857** [reply]
Oh, my bad. I thought you were getting errors building a robot project because of PPLib. This makes a lot more sense now.

Are you running the build command before you run publish? I'm not sure if its needed but I always run both in order, both with `-PreleaseMode`

**Reply by jasonli0616** [reply]
I'm reinstalling WPILib right now, so I'll try that in a sec.

I wasn't running build before, I was just running `./gradlew publish -Preleasemode`. However, just opening the repo in WPILib VSCode was showing errors in Intellisense. If that's not happening for you, it's probably my installation of WPILib

**Reply by mjansen4857** [reply, marked as answer]
I dealt with a lot of errors like this while developing before the 2023 season and it shouldn't need a reinstall. You don't even need WPILib installed at all to develop projects like this since Gradle takes care of all of it. The release mode flag is case sensitive, so make sure it is exactly `-PreleaseMode`

I'm not sure how VSCode handles the intellisense, but it always seems to do weird things for me. The intellisense might be fixed after you get a release mode build working. I usually just use IntelliJ instead of VSCode, it has a much better understanding of Gradle projects and the intellisense is usually more useful.

**Reply by jasonli0616** [reply]
🤦‍♂️🤦‍♂️ Yep it works with `-PreleaseMode`. Gotta ensure that the capitalization was right. Thanks for the help!
